### DISCUSSION TITLE
How is the build optimized with `getStaticPaths` and `getStaticProps`?

### DISCUSSION THREAD

**Discussion by josias-r** [vercel/next.js#11052] [Help] [answered]
If I use graphQL for instance inside the `getStaticPaths` only, will it get bundled into the production anyways even if it's never used somewhere else at runtime?
The same thing for  `getStaticProps` ofc, with the exception, that modules should get loaded when preview mode is entered, right?

**Comment by Timer** [comment, marked as answer]
The code used for `getStaticProps` / `getStaticPaths` is always **eliminated** from the client-side JavaScript files—even in Preview Mode.

This allows you to use Node.js-specific packages, like `fs` or a direct call to a database, without worrying about client-side size or compatibility.

**Reply by josias-r** [reply]
Oh so in preview mode it will actually run directly on the server? That's so cool thx! This will open a lot of doors for me 😛

**Reply by Timer** [reply]
Yup! Preview mode runs directly on the server.

**Reply by josias-r** [reply]
... That would also mean any auth token used in those functions would never be accessible to the public, noice 🥳

**Reply by Timer** [reply]
Correct!

**Reply by Timer** [reply]
You can use this website to see what Next.js will bundle for the client side, given an input:
https://next-code-elimination.now.sh/

**Reply by josias-r** [reply]
Awesome 🎉
